system/security/shared/caching.cpp

```diff
@@ -549,8 +557,6 @@ inline void CPermissionsCache::removeAllManagedFileScopes()
 
     etc. Until full scope path checked, or no read permissions hit on ancestor scope.
 */
-static CriticalSection msCacheSyncCS;//for managed scopes cache synchronization
-static CriticalSection syncDefaultScopePermissions;//for cached default file scope permissions
 bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const char * fullScope, StringBuffer& managedScope, SecAccessFlags * accessFlags)
```

Comment:
question: What happens if permissions are requested before the update thread has started?

Reply:
Also, moved cached timeout to be a constructor parameter. That way the fill thread, and initial fill start, is completed during construction. Any one trying to query before the initial fill is complete will block.